cli/src/semgrep/meta.py

```diff
@@ -99,7 +99,24 @@ def repo_name(self) -> str:
 
     @property
     def repo_url(self) -> Optional[str]:
-        return get_url_from_sstp_url(os.getenv("SEMGREP_REPO_URL"))
+        env = get_state().env
+        repo_url = os.getenv("SEMGREP_REPO_URL")
+        if not repo_url:
+            # if the repo URL was not explicitly provided, try getting it from git
+            # nosem: use-git-check-output-helper
+            git_parse = subprocess.run(
+                ["git", "remote", "get-url", "origin"],
+                capture_output=True,
+                encoding="utf-8",
+                timeout=env.git_command_timeout,
+            )
+            if git_parse.returncode != 0:
+                logger.warn(
+                    f"Unable to infer repo_url. Set SEMGREP_REPO_URL environment variable or run in a valid git project with remote origin defined"
+                )
+            repo_url = git_parse.stdout.strip()
```

Comment:
```suggestion
            repo_url = None
            if git_parse.returncode == 0:
                repo_url = git_parse.stdout.strip()
            else:
                logger.warn(
                    f"Unable to infer repo_url. Set SEMGREP_REPO_URL environment variable or run in a valid git project with remote origin defined"
                )
```
this might be nice since `get_url_from_sstp_url` should explicitly handle being passed `None`

Reply:
Going to keep this consistent with other invocations and leave for a later cleanup that will properly use the git check output helper